npm install fails on CentOS 7

npm install fails installing ioctl with the following error
```
> ioctl@2.0.0 install /home/rahul/S3/node_modules/ioctl
> node-gyp rebuild

make: Entering directory `/home/rahul/S3/node_modules/ioctl/build'
  CXX(target) Release/obj.target/ioctl/src/ioctl.o
make: g++: Command not found
make: *** [Release/obj.target/ioctl/src/ioctl.o] Error 127
make: Leaving directory `/home/rahul/S3/node_modules/ioctl/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 3.10.0-327.18.2.el7.x86_64
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/rahul/S3/node_modules/ioctl
gyp ERR! node -v v4.4.5
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok 
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@1.0.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
```

--- a/init.js
+++ b/init.js
@@ -6,10 +6,16 @@ const fs = require('fs');
 const os = require('os');
 
 const async = require('async');
-const ioctl = require('ioctl');
 const constants = require('./constants').default;
 const config = require('./lib/Config.js').default;
 const logger = require('./lib/utilities/logger.js').logger;
+
+let ioctl;
+try {
+    ioctl = require('ioctl');
+} catch (err) {
+    logger.warn('ioctl dependency is unavailable. skipping...');
+}
 
 function _setDirSyncFlag(path) {
     const GETFLAGS = 2148034049;
@@ -48,7 +54,7 @@ const warning = 'WARNING: Synchronization directory updates are not ' +
     'supported on this platform. Newly written data could be lost ' +
     'if your system crashes before the operating system is able to ' +
     'write directory updates.';
-if (os.type() === 'Linux' && os.endianness() === 'LE') {
+if (os.type() === 'Linux' && os.endianness() === 'LE' && ioctl) {
     try {
         _setDirSyncFlag(dataPath);
         _setDirSyncFlag(metadataPath);

--- a/package.json
+++ b/package.json
@@ -38,6 +38,9 @@
     "xml": "~1.0.0",
     "xml2js": "~0.4.12"
   },
+  "optionalDependencies": {
+      "ioctl": "2.0.0"
+  },
   "devDependencies": {
     "babel-cli": "^6.2.0",
     "babel-eslint": "^6.0.0",

